### PR TITLE
Fix for Issue #49

### PR DESCRIPTION
**Pull request by uromahn**
Adding status test for pods before adding them.
In case you accept this PR, please issue a new release with version 1.0.7 so we can pick it up.
In case you don't accept, please let me know what should change.

### CODE REVIEW

File: src/main/java/org/jgroups/protocols/kubernetes/Client.java [hunk 1]
```diff
@@ -104,20 +104,91 @@ protected String fetchFromKubernetes(String op, String namespace, String labels,
...
+        log.trace("Determining pod status");
+        String phase = Optional.ofNullable(podStatus.at("phase"))
+                .map(Json::asString)
+                .orElse("not running");
+        log.trace("  status.phase=%s", phase);
+        if(!phase.equalsIgnoreCase("Running")) {
+            return false;
+        }
+        // 2. and 3. status.message and status.reason
+        String statusMessage = Optional.ofNullable(podStatus.at("message"))
```
Comment (slaskawi) [inline src/main/java/org/jgroups/protocols/kubernetes/Client.java]: It seems you're extracting a String value here and then just check if it exists. Would it be possible to use `has` method?

File: src/main/java/org/jgroups/protocols/kubernetes/Client.java [hunk 2]
```diff
@@ -104,20 +104,91 @@ protected String fetchFromKubernetes(String op, String namespace, String labels,
...
+                .orElse(null);
+        String statusReason = Optional.ofNullable(podStatus.at("reason"))
+                .map(Json::asString)
+                .orElse(null);
+        log.trace("  status.message=%s and status.reason=%s", statusMessage, statusReason);
+        if(statusMessage != null || statusReason != null) {
+            return false;
+        }
+        // 4. status.containerStatuses.ready
+        List<Json> containerStatuses = Optional.ofNullable(podStatus.at("containerStatuses"))
```
Comment (slaskawi) [inline src/main/java/org/jgroups/protocols/kubernetes/Client.java]: Here you could go through the list of containers and check which one is not ready. If the list is not empty, the pod is not ready.

File: src/main/java/org/jgroups/protocols/kubernetes/Client.java [hunk 3]
```diff
@@ -104,20 +104,91 @@ protected String fetchFromKubernetes(String op, String namespace, String labels,
...
+            return false;
+        }
+        // 5. ready condition must be "True"
+        Boolean readyCondition = Boolean.FALSE;
+        List<Json> conditions = podStatus.at("conditions").asJsonList();
+        // walk through all the conditions and find type=="Ready" and get the value of the status property
+        for(Json condition: conditions) {
+            String type = condition.at("type").asString();
+            if(type.equalsIgnoreCase("Ready")) {
+                readyCondition = new Boolean(condition.at("status").asString());
```
Comment (slaskawi) [inline src/main/java/org/jgroups/protocols/kubernetes/Client.java]: Using `Boolean#valueOf` might be slightly more efficient here.

**Comment by belaban** [comment]
@slaskawi If you think the PR is good, I'm fine, too

**Comment by slaskawi** [comment]
Integrated, thanks @uromahn!

**Pull request merged**